core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -315,15 +316,30 @@ class UserService(
   def mapCloudPlatformAndPolicies(
     project: RawlsBillingProject,
     billingProfile: Option[ProfileModel],
-    roles: Set[ProjectRole]
+    roles: Set[ProjectRole],
+    workspaceManagerDao: WorkspaceManagerDAO
   ): RawlsBillingProjectResponse = (project.billingProfileId, billingProfile) match {
     case (None, _) => RawlsBillingProjectResponse(roles, project, CloudPlatform.GCP, protectedData = None)
     case (Some(_), Some(p)) =>
       val platform = CloudPlatform(p)
-      val responseProject = if (platform == CloudPlatform.AZURE) {
+      val (responseProject, maybeRegion) = if (platform == CloudPlatform.AZURE) {
         val c = AzureManagedAppCoordinates(p.getTenantId, p.getSubscriptionId, p.getManagedResourceGroupId)
-        project.copy(azureManagedAppCoordinates = Some(c))
-      } else project
+        val projectWithCoords = project.copy(azureManagedAppCoordinates = Some(c))
+        val landingZoneRegion = project.landingZoneId match {
+          case Some(landingZoneId) =>
+            Try(workspaceManagerDao.getLandingZone(UUID.fromString(landingZoneId), ctx)) match {
+              case Success(landingZone) => Some(landingZone.getRegion)
```

Comment:
If you use `Option(landingZone.getRegion)`, you'll get the result of `None` if it's absent/null, and `Some(region) if present.
That's probably preferable to the behavior as written here, which will give `Some(null)` if the region is not present, since nulls should generally be avoided in scala (avoiding null anywhere is ideal IMO, but scala has explicit conventions around discouraging null values).

Reply:
Ah, definitely better. Thank you!